src/grisp_bitmap.erl

```diff
@@ -8,25 +12,70 @@
 -export([pp/1]).
 -export([pp/2]).
 
+%--- Types ---------------------------------------------------------------------
+-type coding() :: bin | binary | dec | decimal | hex | hexadecimal
+                  | nib | nibble.
+
 %--- API -----------------------------------------------------------------------
 
+%% @doc Replace a part of a bitsting.
+%%
+%% === Example ===
+%% ```
+%%  5> grisp_bitmap:set_bits(<<1,2,3,4,5>>, 6, <<2:2>>).
+%%  <<2,2,3,4,5>>
+%% '''
+-spec set_bits(bitstring(),non_neg_integer(),bitstring()) -> bitstring().
 set_bits(Bin, Start, Value) when bit_size(Bin) >= Start + bit_size(Value) ->
     Len = bit_size(Value),
     <<Prefix:Start/bitstring, _:Len/bitstring, Postfix/bitstring>> = Bin,
     <<Prefix/bitstring, Value/bitstring, Postfix/bitstring>>.
 
+%% @doc Get a part of a bitstring.
+%%
+%% === Example ===
+%% ```
+%%  1> grisp_bitmap:get_bits(<<1,2,3,4,5>>, 6, 2).
+%%  <<1:2>>
+%% '''
+-spec get_bits(bitstring(),non_neg_integer(),non_neg_integer()) -> bitstring().
 get_bits(Bin, Start, Len) when bit_size(Bin) >= Start + Len ->
     <<_:Start/bitstring, Bytes:Len/bitstring, _/bitstring>> = Bin,
     Bytes.
 
+%% @equiv set_bits(Bin, Start * 8, Value)
+-spec set_bytes(bitstring(),non_neg_integer(),bitstring()) -> bitstring().
 set_bytes(Bin, Start, Value) when byte_size(Bin) >= Start + byte_size(Value) ->
     set_bits(Bin, Start * 8, Value).
 
+%% @equiv get_bits(Bin, Start * 8, Len * 8)
+-spec get_bytes(bitstring(),non_neg_integer(),non_neg_integer()) -> bitstring().
 get_bytes(Bin, Start, Len) when byte_size(Bin) >= Start + Len ->
     get_bits(Bin, Start * 8, Len * 8).
 
+%% @doc Print bitstring as hexadecimal numbers.
```

Comment:
This actually only prints `binary()` and not `bitstring()`, which is probably a bug 😄